src/bridge-canister/src/operation_store.rs

```diff
@@ -200,21 +211,40 @@ where
         &self,
         dst_address: &H160,
         pagination: Option<Pagination>,
+        memo: Option<String>,
```

Comment:
The name of the method `get_for_address` suggests that it returns all operation related to the given `dst_address`, but instead what new implementation does is returning one operation with given memo if `memo` is given or return all operations for the address if `memo` not given. Even this explanation sounds like bad API :-).

So I would suggest not add the new parameter to this method, and just have a separate method below.